In [2]:
import pandas as pd

In [3]:
datos=pd.read_csv("synergy_logistics_database.csv") #Lectura de archivo csv
datos #Muestra de datos

register_id direction    origin destination  year      date  \
0                1   Exports     Japan       China  2015  31/01/15   
1                2   Exports     Japan       China  2015  01/02/15   
2                3   Exports     Japan       China  2015  02/02/15   
3                4   Exports     Japan       China  2015  03/02/15   
4                5   Exports     Japan       China  2015  04/02/15   
...            ...       ...       ...         ...   ...       ...   
19051        19052   Imports     Japan   Singapore  2020  27/06/20   
19052        19053   Imports  Malaysia   Singapore  2020  28/06/20   
19053        19054   Imports  Malaysia   Singapore  2020  29/06/20   
19054        19055   Imports  Malaysia   Singapore  2020  30/06/20   
19055        19056   Imports  Malaysia   Singapore  2020  01/07/20   

            product transport_mode     company_name  total_value  
0              Cars            Sea            Honda     33000000  
1              Cars            Sea            Honda     16000000  
2              Cars            Sea            Honda     29000000  
3              Cars            Sea            Honda     14000000  
4              Cars            Sea            Honda     17000000  
...             ...            ...              ...          ...  
19051  Gas turbines            Sea  Union Energy Co      1000000  
19052  Gas turbines            Sea  Union Energy Co      2000000  
19053  Gas turbines            Sea  Union Energy Co     33000000  
19054  Gas turbines            Sea  Union Energy Co     13000000  
19055  Gas turbines            Sea  Union Energy Co     30000000  

[19056 rows x 10 columns]

In [4]:
Total_Empresa=datos['total_value'].sum() #Suma del valor de todas las importaciones y exportaciones
Total_Empresa 

215691298000

# Opción 1: Revisar las rutas de importación y exportación

Opción 1) Rutas de importación y exportación. Synergy logistics está
considerando la posibilidad de enfocar sus esfuerzos en las 10 rutas más
demandadas. Acorde a los flujos de importación y exportación, ¿cuáles son esas
10 rutas? ¿le conviene implementar esa estrategia? ¿porqué? 


In [5]:
#Creacion de dataframe que contiene todas las rutas de importacion y exportación
rutas=datos.copy().groupby(by=['origin','destination']).size().reset_index().rename(columns={0:'count'}) 
#rutas es la lista de todas las rutas dentro del archivo csv
rutas1=datos.copy() #Copia de los datos
#Descripcion_rutas=rutas.describe()
#type(rutas)
#Muestra las 15 rutas mas usadas dentro de la empresa


In [6]:
total_ruta=list() #Lista que contendra origen, destino y la suma de todas las transacciones en esa ruta
for i in range(len(rutas)):
    origen=rutas.iloc[i]['origin'] #dada la fila i, desplagaremos el valor origin
    destino=rutas.iloc[i]['destination'] #dada la fila i, desplagaremos el valor destination
    celdas=rutas1[rutas1['origin'].str.contains(origen) & rutas1['destination'].str.contains(destino)] #Haremos una busqueda de todas las filas que provengan de la ruta deseada
    total=celdas['total_value'].sum() #Suma de las celdas total_value elegidas por la linea anterior
    #total_ruta.append([origen,destino,total])
    total_ruta.append([origen,destino,total]) #Agregamos la ruta y el total que fue movido en esas rutas
    #rutas.iloc[0]['origin']

In [16]:
print(rutas1[rutas1['origin'].str.contains('China') & rutas1['destination'].str.contains('Mexico')]['total_value'].sum()/Total_Empresa)

0.05792537814854264


In [21]:
#Transformamos la lista de origen,destino y total en un dataframe.
total_por_rutas=pd.DataFrame(total_ruta)
total_por_rutas.columns=['origen','destino','total'] #renombramos las columnas del dataframe.
total_por_rutas=total_por_rutas.sort_values(by=['total'],ascending=False).head(10) #ordenamos las rutas por el total, de menor a mayor. Y mostramos las 10 mas importantes 
total_por_rutas

origen         destino        total
34         China          Mexico  12494000000
26        Canada          Mexico   8450000000
138  South Korea         Vietnam   6877007000
33         China           Japan   5891000000
90         Japan          Mexico   5829000000
43        France         Belgium   5538069000
53        France  United Kingdom   5427000000
37         China     South Korea   4790000000
134  South Korea           Japan   4741000000
158          USA          Mexico   4710000000

In [25]:
print(total_por_rutas['total'].head(10).sum(),Total_Empresa) #Proporcion del flujo de rutas mas transitadas entre el total de la empresa

64747076000 215691298000


El top 10 de rutas maneja el 30.01% del dinero que maneja la empresa

# Tipos de transporte


Opción 2) Medio de transporte utilizado. ¿Cuáles son los 3 medios de transporte
más importantes para Synergy logistics considerando el valor de las
importaciones y exportaciones? ¿Cuál es medio de transporte que podrían
reducir?

In [26]:
#Creacion de dataframe que contiene cuantas veces se usa un medio de transporte
transporte=datos.copy().groupby(by=['transport_mode']).size().reset_index().rename(columns={0:'count'}) 
#Muestra de cuantas veces se usa cada tipo de transporte
transporte

transport_mode  count
0            Air   2389
1           Rail   3381
2           Road   2598
3            Sea  10688

In [27]:
total_transporte=list() #Lista que incluira tipo de transporte y el total manejado por ese tipo de transporte
for i in range(len(transporte)):
    via_transporte=transporte.iloc[i]['transport_mode'] #localizamos el valor 'transpor_mode' de la fila i del DataFrame transporte
    celdas=datos[datos['transport_mode'].str.contains(via_transporte)] #Buscamos todas las filas que cumplan con la condicion transpor_mode==transporte
    total=celdas['total_value'].sum()#suma de total_value de  todas las celdas elegidas
    #total_ruta.append([origen,destino,total])
    total_transporte.append([via_transporte,total]) #añadimos lista via_transporte, total

In [28]:
total_por_transporte=pd.DataFrame(total_transporte) #transformamos la lista total_transporte en un dataframe

total_por_transporte.columns=['transporte','total'] #renombramos las columnas a transporte y total
total_por_transporte.sort_values(by=['total'],ascending=False) #Ordenamos de mayor a menor

transporte         total
3        Sea  100530622000
1       Rail   43628043000
0        Air   38262147000
2       Road   33270486000

In [29]:
total_por_transporte['total'][2]

33270486000

In [30]:
print(total_por_transporte['total'][2]/total_por_transporte['total'].sum())

0.1542504788487109


In [31]:
paises_road=datos[datos['transport_mode'].str.contains('Road')]
print(len(paises_road.groupby(by=['destination']).size().reset_index().rename(columns={0:'count'})))
print(len(paises_road.groupby(by=['origin']).size().reset_index().rename(columns={0:'count'})))


16
13


Reducir el flujo de en transporte terrestre solo involucra al 15.42%  de todo el dinero manejado dentro de la empresa
Pero reducir ese flujo involucra hacer cambios en 13 paises de origen y 16 de destino

# Valor total de importaciones y exportaciones

Si Synergy Logistics
quisiera enfocarse en los países que le generan el 80% del valor de las
exportaciones e importaciones ¿en qué grupo de países debería enfocar sus
esfuerzos?


In [32]:
valor_corte=Total_Empresa*0.8 #Valor de corte, es el valor del 80$ de todas las importaciones y exportaciones

In [33]:
#Dataframe con todos los paises de origen de una importacion y exportacion
paises=datos.copy().groupby(by=['origin']).size().reset_index().rename(columns={0:'count'})
paises 

origin  count
0              Australia    965
1                Austria    437
2                Belgium    243
3                 Brazil    515
4                 Canada    650
5                  China   2272
6                 France   1088
7                Germany   1664
8                  India    565
9                  Italy    648
10                 Japan   1702
11              Malaysia    199
12                Mexico   1138
13           Netherlands    796
14                Russia    808
15             Singapore    537
16           South Korea   1295
17                 Spain    478
18           Switzerland    249
19                   USA   1840
20  United Arab Emirates     20
21        United Kingdom    929
22               Vietnam     18

In [34]:
total_paises=list()
for i in range(len(paises)):
    origen=paises.iloc[i]['origin']
    #direccion=paises.iloc[i]['direction']
    celdas=datos[datos['origin'].str.contains(origen)]#& datos['direction'].str.contains(direccion)]
    sumas_total=celdas['total_value'].sum()
    total_paises.append([origen,sumas_total]) #Lista con todos los paises de origen y cuanto dinero ha sido manejado por importaciones y exportaciones


In [37]:
#datos[datos['origin'].str.contains('Austria') &datos['direction'].str.contains('Imports') ]

In [36]:
#Dataframe con los paises con origen y total de todas las importaciones y exportaciones
Importacion_y_exportacion=pd.DataFrame(total_paises,columns=[ 'origen','total']).sort_values(by=['total'],ascending=False).reset_index()
#Importacion_y_exportacion
total_acumulado_pais=0 #variable a iterar
for i in range(len(Importacion_y_exportacion)):
    total_acumulado_pais+=Importacion_y_exportacion['total'][i] #suma de los totales de exportacione e importacion de cada pais
    if total_acumulado_pais>valor_corte: #cuando se llegue al pais que haga que la suma sea mayor al 80% del valor total de la empresa se detiene
        break
#print(total_acumulado_pais/Total_Empresa,i)
Importacion_y_exportacion.head(i+1) #Imprime a los paises que involucran dan el 80% de importaciones y exportaciones

index       origen        total
0      5        China  45210046000
1     19          USA  23646306000
2     10        Japan  20042976000
3      6       France  19930332000
4     16  South Korea  18510146000
5      7      Germany  15593233000
6     14       Russia  14074000000
7      4       Canada  11253000000
8      9        Italy   6634684000

Por totales de importacion y exportacion y  del pais de origen, la empresa hace sus negocios principalmente en 9 paises

In [42]:
Importacion_y_exportacion['origen'][i]

'Italy'

In [43]:
total_acumulado_pais/Total_Empresa

0.8108566484680342

9 países generan el 81% del flujo de importaciones y exportaciones 